plugins/experimental/carp/CarpConfig.cc

```diff
@@ -0,0 +1,581 @@
+ */
+
+//////////////////////////////////////////////////////////////
+// Read CARP configuration file
+// [Servers]
+// host1.yahoo.com:4080 weight=2      # port 4080 on host1.yahoo.com with weight factor of 2
+// host2.yahoo.com                    # port 80 on host2.yahoo.com with (default) weight factor of 1
+// 
+// [Values]
+// healthcheck={host}:8001/status.html
+// healthfreq=30
+// global=on
+//
+
+#include <stdlib.h> 
+#include <stdio.h> 
+#include <memory.h> 
+#include <ts/ts.h>
+#include <sys/time.h>
+
+#include <sstream>
+
+#include "CarpConfig.h"
+#include "Common.h"
+#include "CarpConfigPool.h"
+
+using namespace std;
+
+#define DEFAULT_HEALTH_CHECK_FREQ 30  // 30 second period for health checks
+#define DEFAULT_HEALTH_CHECK_PORT 80  // default to makeing healthcheck requests against port 80
+#define DEFAULT_CONFIG_RELOAD_FREQ 30 // 30 seconds used in TSContSchedule
+#define DEFAULT_PORT 80  // default to makeing requests against port 80
+#define DEFAULT_WEIGHT 1  // default weight
+#define DEFAULT_SCHEME "http"
+#define DEFAULT_REPLICATION_FACTOR 1
+
+// config section headers
+static const char *const SECTION_SERVERS_STR = "[Servers]";
+static const char *const SECTION_VALUES_STR = "[Values]";
+
+// key strings
+static const char *const KEY_HEALTHCHECK_STR = "healthcheck";
+static const char *const KEY_HEALTHFREQ_STR = "healthfreq";
+static const char *const KEY_RELOADFREQ_STR = "reloadfreq";
+static const char *const KEY_HCTIMEOUT_STR =  "hctimeout";
+static const char *const KEY_BLACKLIST_STR = "blacklist";
+static const char *const KEY_WHITELIST_STR = "whitelist";
+static const char *const KEY_MODE_STR = "mode";
+static const char *const KEY_ALLOWFWDPORT_STR = "allowfwdport";
+static const char *const KEY_REPLICATIONFACTOR_STR = "replicationfactor";
+
+// parameter strings
+static const char *const WEIGHT_EQUALS_STRING = "weight=";
+static const char *const GROUP_EQUALS_STRING = "group=";
+static const char *const KEY_MODE_PREREMAP_STR = "pre-remap";
+static const char *const KEY_MODE_POSTREMAP_STR = "post-remap";
+
+
+/**********************************************************/
+bool
+getInt(char** pptr, int *val)
+{
+  bool bReturn = false;
+  int v=0;
+  
+  char* ptr = *pptr;
+  // skip white space if any
+  while (*ptr && isspace(*ptr)) ++ptr; 
+  // get digits
+  if (*ptr) {
+    while (*ptr && isdigit(*ptr)) {
+      v *= 10;
+      v += (*ptr)-'0';
+      ++ptr;
+      bReturn = true;
+    }
+  }
+  if (bReturn) {
+    *pptr = ptr;
+    *val = v;
+  }
+
+  return bReturn;
+}
+
+/**********************************************************/
+// [http[s]://]host[:port]/path
+bool
+getHostAndPort(char** pptr,string* sHost,int* iPort, string* sScheme)
+{
+  bool bReturn = false;
+  char* ptr = *pptr;
+
+  *iPort = 80;
+  *sScheme = DEFAULT_SCHEME;
+  if(strncmp(*pptr,"https",5) == 0) {
+    *iPort = 443;
+    *sScheme = "https";
+  }
+    
+  //skip leading white space
+  while (*ptr && isspace(*ptr)) ++ptr;
```

Comment:
AFAICT, "https://" and "  https://" will behave differently because the port is set to 443 only if the string immediately begins with "https". Is that intended? Why check that twice in any case, instead of handling the port in the check just below?